docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+}
+```
+
+Once implemented the agent may register the structure that implements it with the following call.
+
+```go
+import "github.com/Axway/agent-sdk/pkg/agent"
+
+agent.RegisterProvisioner(provisioningImplementation)
+```
+
+#### RequestStatus return structure
+
+All Provisioning request methods will expect a RequestStatus object to be returned after handing the request.  To create this object the Agent SDK provides a builder with the following methods.
+
+```go
+// RequestStatusBuilder - builder to create new request Status
+type RequestStatusBuilder interface {
+  // SetMessage - set the request Status message
+  SetMessage(message string) RequestStatusBuilder
+  // SetProperties - set the properties of the RequestStatus
+  SetProperties(map[string]string) RequestStatusBuilder
+  // AddProperty - add a new property on the RequestStatus
+  AddProperty(key string, value string) RequestStatusBuilder
+  // Success - set the status as success
+  Success() RequestStatus
+  // Failed - set the status as failed
+  Failed() RequestStatus
+}
+```
+
+The properties that are added to the request status are persisted back to the corresponding resource and will be available for subsequent provisioning/deprovisioning requests for that resource.
+
+Example of a successful provisioning request that adds one property to the status.
+
+```go
+return provisioning.NewRequestStatusBuilder().AddProperty("ApplicationID", "apiid").Success()
+```
+
+Example of a failed provisioning request that adds a failure message.
+
+```go
+return provisioning.NewRequestStatusBuilder().SetMessage("failed for reason x").Failed()
+```
+
+### Application Request Provision and Deprovision
+
+The Application request flow begins with the creation of a ManagedApplication resource in the environment that the agent is managing. The agent SDK will receive an event that the resource was created then check that the resource is in a Pending state before handing it off to the agent. Below is an event diagram to show the process taken.
+
+![Application Events](./managedapp.png)
+
+Both provision and deprovision requests will receive an ApplicationRequest object that contains the following methods.
+
+```go
+// ApplicationRequest - interface for agents to use to get necessary application request details
+type ApplicationRequest interface {
+  // GetManagedApplicationName returns the name of the managed application on this request
+  GetManagedApplicationName() string
+  // GetApplicationDetailsValue returns a value found on the 'x-agent-details' sub resource of the ManagedApplication
+  GetApplicationDetailsValue(key string) string
+}
+```
+
+When receiving a provisioning request the agent is expected to create, on its connected data plane, an Application (or similar object), that will be associated with APIs or Credentials. During this provisioning the agent should gather any details necessary to handle deprovisioning and set it in the [RequestStatus](#requeststatus-return-structure).
```

Comment:
Lost on this first sentence..  like its missing the verb when i just read it... 

on the second sentence, we need another comma break after provisioning
During this provisioning, the agent should gather any details necessary to handle deprovisioning